deploy/examples/create-external-cluster-resources.py

```diff
@@ -632,7 +636,10 @@ def __init__(self, arg_list=None):
         if not self._arg_parser.rgw_pool_prefix and not self._arg_parser.upgrade:
             self._arg_parser.rgw_pool_prefix = self.DEFAULT_RGW_POOL_PREFIX
         if self.ceph_conf:
-            self.cluster = rados.Rados(conffile=self.ceph_conf)
+            kwargs = {}
+            if self.ceph_keyring:
+                kwargs["conf"].update({"keyring": self.ceph_keyring})
```

Comment:
have you tested is this work as expected,
Originally it say:
`  cluster = rados.Rados(conffile = 'ceph.conf', conf = dict (keyring = '/path/to/keyring'))`

Reply:
yes, if a keyring is specified, kwargs would be `{"conf": {"keyring": path}}`, otherwise empty, the calls thus become:

```py
# note dict(keyring="data") == {"keyring": "data"}

# with keyring
rados.Rados(conffile=conf, conf=dict(keyring=keyring))

# without keyring
rados.Rados(conffile=conf)
```

We could always pass a `conf`, perhaps as an empty dict, but I wanted to avoid unnecessary conflicts in case of defaults that might be overriden from an empty config.